### PR TITLE
File Input/Output in C++

### PR DESCRIPTION
**Pull request by Noah11012**
## Please Complete the Following

- [ ] I would **NOT** like credit for my contribution in the article (credit will be included otherwise)
- [ ] I plan to write the article for this code snippet
- [x] I named the pull request using `<Sample Program> in <Language>` format
- [x] I created/updated the language README
  - [x] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut, developer, typing, etc.)
  - [x] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][2])
- [ ] I fixed #your-issue-number-here

### CODE REVIEW

**Comment by jrg94** [comment]
This looks great! Now, I want to chat about a couple things. 

1. We need to formalize this in the contributing doc.
2. We should probably remove those comments.

I'll address both:

**First**, would you mind writing up something more formal/generic that we could put into the contributing doc? That way others can try to reproduce your solution in another language. 

**Second**, you have documented your solution quite a bit. I like to keep these snippets free of comments and supplement them with articles. Would you want to write the article for this solution?

**Comment by Noah11012** [comment]
>First, would you mind writing up something more formal/generic that we could put into the contributing doc? That way others can try to reproduce your solution in another language.

Sorry if I'm being slow to understand, but I'm not sure what you mean by "contributing doc". Do you mean the CONTRIBUTING.md file in the root directory of the repository or a separate new contribution document in the c-plus-plus folder?

>Second, you have documented your solution quite a bit. I like to keep these snippets free of comments and supplement them with articles. Would you want to write the article for this solution?

I've never really written an article before but I wouldn't mind trying.

**Comment by jrg94** [comment]
No worries! 

Yes, I mean the contributing.md doc that I linked in our discussion on #403. If you could create a new section for File IO, that would be awesome. We'll walk through the wording when it's ready, so don't worry about it.

The articles aren't too bad! They're about 500 words, and we have a regular format. If you can follow it, we'll be good.

**Comment by Noah11012** [comment]
I'll write the article then. Is it for general file I/O or for C++ specifically?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -262,6 +262,15 @@ GUI elements. Keep it simple. Remember, the goal is to show off language feature
 Also, I ask that you don't use external libraries. I like for these files to
 be as easy as possible to test, so limiting dependencies is helpful.
 
+### File Input/Output
```
Comment (jrg94) [inline CONTRIBUTING.md]: This explains what file IO is, but it doesn't outline what exactly the program should accomplish. In other words, is there a specific thing you'd like the program to accomplish?

Based on your example, I assume we write some arbitrary text to a file and read it back.

**Comment by jrg94** [comment]
It would be for C++. You can take a look at an example:

- [Reverse a String in Python](https://therenegadecoder.com/code/reverse-a-string-in-python/)

**Comment by Noah11012** [comment]
Could you create a series for File I/O in every language? I don't see the option in the Add New Post webpage.

**Comment by jrg94** [comment]
Sure! Are you planning to write the first article as well (File IO in Every Language) or just the C++ one?

**Comment by Noah11012** [comment]
For now I plan only for the C++ one. But may expand do it for other languages like Python.

**Comment by jrg94** [comment]
Sounds good! Thanks for your help 😄

**Comment by jrg94** [comment]
Alright, I edited the directions a bit. Let me know if you like the changes, and I'll merge them in.

**Comment by Noah11012** [comment]
Everything looks good to me.

**Pull request merged**